include/micm/util/matrix.hpp

```diff
@@ -50,7 +52,7 @@ namespace micm
         // check that this row matches the expected rectangular matrix dimensions
         if (other.size() < y_dim_)
         {
-          throw std::runtime_error("Matrix row size mismatch in assignment from vector");
+          throw std::system_error(make_error_code(MicmMatrixErrc::RowSizeMismatch), "");
```

Comment:
Might be worth having a message why it's mismatched for easier debugging?

Reply:
done!